### PR TITLE
[SPARK-12107] [EC2] Update spark-ec2 versions

### PR DESCRIPTION
**Pull request by nchammas**
I haven't created a JIRA. If we absolutely need one I'll do it, but I'm fine with not getting mentioned in the release notes if that's the only purpose it'll serve.

cc @marmbrus - We should include this in 1.6-RC2 if there is one. I can open a second PR against branch-1.6 if necessary.

### CODE REVIEW

File: ec2/spark_ec2.py [hunk 1]
```diff
@@ -87,7 +90,10 @@
     "1.3.1": "0.5.0",
     "1.4.0": "0.6.4",
     "1.4.1": "0.6.4",
-    "1.5.0": "0.7.1"
+    "1.5.0": "0.7.1",
+    "1.5.1": "0.7.1",
+    "1.5.2": "0.7.1",
+    "1.6.0": "0.7.1",
```
Comment (nchammas) [inline ec2/spark_ec2.py]: I'm not sure about this mapping. I'm just assuming that a version of Tachyon that works with 1.5.0 will also work with later versions. @haoyuan?

**Comment by marmbrus** [comment]
JIRA please.

**Comment by nchammas** [comment]
:ok_hand:

**Comment by nchammas** [comment]
Done.

File: ec2/spark_ec2.py (hunk 1 above)
Comment (JoshRosen) [inline ec2/spark_ec2.py]: FWI we're now depending on Tachyon 0.8.2.

Comment (nchammas) [inline ec2/spark_ec2.py]: Starting with Spark 1.6.0?

Comment (JoshRosen) [inline ec2/spark_ec2.py]: Yes.

Comment (nchammas) [inline ec2/spark_ec2.py]: Done.

**Comment by shivaram** [comment]
LGTM. Merging this. Thanks @nchammas

**Comment by shivaram** [comment]
Well there is one issue here. The launch will fail for users on master, branch-1.6 until the 1.6.0 tar balls are available on S3. I am not sure there is a nice way around this.
One way around is to only merge this on branch-1.6 and keep the PR open and merge it on master after 1.6 is released.

Thoughts ?

**Comment by nchammas** [comment]
What have we done in the past in this situation?

People running on master can get around this by manually specifying the Spark version with `-v` until 1.6.0 is published. Is that acceptable?

**Comment by shivaram** [comment]
I think in the past our approach has been

1. Not update the `DEFAULT_SPARK_VERSION` on master until the release is up
2. During the release tag, @pwendell used to checkin a PR to update the `DEFAULT_SPARK_VERSION` on the release branch (kind of similar to how the maven pom versions are set). 
3. Also backport the PR from step 1 to `branch-1.x` after the release.

This is obviously very cumbersome (especially step 2 ?) and I'm not sure if @pwendell has any scripts which automate the release step especially.

**Comment by nchammas** [comment]
If it helps simplify things, why don't we immediately merge in this change to both master and branch-1.6, and simply ask people to use `-v` to specify a different version of Spark until 1.6 is actually published?

I think it's reasonable to expect people running on the bleeding edge to deal with really minor annoyances like this if means making life easier for the maintainers.

**Comment by shivaram** [comment]
Alright I'm fine with merging this since (a) we are pretty close to the release and (b) we should try to pull out spark_ec2.py from the apache repo to avoid this for the 2.0 release.

**Comment by nchammas** [comment]
:+1:

**Pull request closed without merging**